src/cloud_auth.ts

```diff
@@ -7,51 +7,69 @@ import { User } from './config_types';
 
 export class CloudAuth implements Authenticator {
     public isAuthProvider(user: User): boolean {
-        return user.authProvider.name === 'azure' ||
-            user.authProvider.name === 'gcp';
+        return (
+            user.authProvider.name === 'azure' ||
+            user.authProvider.name === 'gcp'
+        );
     }
 
     public getToken(user: User): string | null {
         const config = user.authProvider.config;
-        // This should probably be extracted as auth-provider specific plugins...
-        let token: string = 'Bearer ' + config['access-token'];
+        if (this.isExpired(config)) {
+            this.updateAccessToken(config);
+        }
+        return 'Bearer ' + config['access-token'];
+    }
+
+    private isExpired(config) {
+        const token = config['access-token'];
         const expiry = config.expiry;
+        if (!token) {
```

Comment:
nit: inline `config['access-token']` or use `contains`

Reply:
would break the symetry here between `access-token` and `expiry`.

i can inline it if you really want, but it does not make the code any better imho